docs/streams/developer-guide/dsl-api.html

```diff
@@ -3609,6 +3631,52 @@ <h5><a class="toc-backref" href="#id34">KTable-KTable Foreign-Key
+                and <code class="docutils literal"><span class="pre">reduce</span></code> operations as well, in addition to
+                <code class="docutils literal"><span class="pre">aggregate</span></code> operations.</li>
+            <li>Table joins will use the latest-by-timestamp record for each key, instead of the latest-by-offset record. Out-of-order updates (per key) will not
+                trigger a new join result. This is true for both primary-key table-table joins and also foreign-key table-table joins. If a
+                versioned table is joined with an unversioned table, the result will be the join of the latest-by-timestamp record from the versioned table with
+                the latest-by-offset record from the unversioned table.</li>
+            <li>Table filter operations will no longer suppress consecutive tombstones, so users may observe more <code class="docutils literal"><span class="pre">null</span></code>
+                records downstream of the filter than compared to when filtering an unversioned table. This is done in order to preserve a complete version history downstream,
+                in the event of out-of-order data.</li>
+            <li><code class="docutils literal"><span class="pre">suppress</span></code> operations are not allowed on versioned tables, as this would collapse the version history
+                and lead to undefined behavior.</li>
+        </ul>
+        <p>Once a table is materialized with a versioned store, downstream tables are also considered versioned until any of the following occurs:</p>
+        <ul class="simple">
+            <li>A downstream table is explicitly materialized, either with an unversioned store supplier or with no store supplier (all stores are unversioned by default, including the default store supplier)</li>
+            <li>Any stateful transformation occurs, including aggregations and joins</li>
+            <li>A table is converted to a stream and back.</li>
+        </ul>
+        <p>The results of certain processors should not be materialized with versioned stores, as these processors do not produce a complete older version history,
+            and therefore materialization as a versioned table would lead to unpredictable results:</p>
+        <ul class="simple">
+            <li>Aggregate processors, for both table and stream aggregations. This includes <code class="docutils literal"><span class="pre">aggregate</span></code>,
+                <code class="docutils literal"><span class="pre">count</span></code> and <code class="docutils literal"><span class="pre">reduce</span></code> operations.</li>
+            <li>Table-table join processors, including both primary-key and foreign-key joins.</li>
+        </ul>
+        <p>For more on versioned stores and how to start using them in your application, see <a href="/{{version}}/documentation/streams/developer-guide/processor-api.html#versioned-state-stores">here</a>.</p>
+    </div>
```

Comment:
Should we mention global stores, too?

Reply:
The section that this line links to (processor API topic on versioned stores) mentions that global tables are not allowed to be materialized with versioned stores. I think since most of the nuances/details are only in the processor API topic, and the DSL API topic links there, it should be OK as is?